test/lib/ansible_test/_data/sanity/pylint/plugins/deprecated.py

```diff
@@ -64,6 +85,20 @@ def _get_expr_name(node):
         return node.func.expr.name
 
 
+def parse_isodate(value):
+    msg = 'Expected ISO 8601 date string (YYYY-MM-DD)'
+    if not isinstance(value, string_types):
+        raise ValueError(msg)
+    # From Python 3.7 in, there is datetime.date.fromisoformat(). For older versions,
+    # we have to do things manually.
+    if not re.match('^[0-9]{4}-[0-9]{2}-[0-9]{2}$', value):
+        raise ValueError(msg)
```

Comment:
This logic appears to be redundant. The exception handler below should catch the same cases the regex does and return the same exception.

Reply:
The code below also accepts `2020-1-1`, which is not ISO 8601. It does catch invalid dates such as `2019-02-29` though, which this regex won't catch.